### PR TITLE
Introduce prometheus to go-ibft

### PR DESCRIPTION
**Pull request by Nemanja0x**
# Description

Tracking time intervals of round and sequence by using histograms.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have added sufficient documentation in code

### CODE REVIEW

File: core/ibft.go [hunk 1]
```diff
@@ -102,13 +103,18 @@ type IBFT struct {
 
 	// validatorManager keeps quorumSize and voting power information
 	validatorManager *ValidatorManager
+
+	// metirc data used to create histograms
```
Comment (Stefan-Ethernal) [inline core/ibft.go]:
```suggestion
	// metric data used to create histograms
```

Comment (Stefan-Ethernal) [reply core/ibft.go]: Fixed https://github.com/0xPolygon/go-ibft/pull/82/commits/abfcc712be3b658d1d9e68eb421986fd608bf072

File: .golangci.yml [hunk 2]
```diff
@@ -83,6 +83,7 @@ linters-settings:
 
   goimports:
     local-prefixes: github.com/0xPolygon/go-ibft
+                    github.com/prometheus/client_golang/prometheus
```
Comment (vcastellm) [inline .golangci.yml:86]: Why is this necessary?

Comment (Stefan-Ethernal) [reply .golangci.yml:86]: Not sure why do we have local imports in the first place... But if we want to keep it, then we must expand that list, because of prometheus-client dependency being added.

**Review by vcastellm — CHANGES_REQUESTED** [review]
Hmmm I'd say to use https://github.com/hashicorp/go-metrics as the metrics lib, then the app using the lib can provide the instance and use any sink it's interested in. Do not use Prometheus lib directly.
